### PR TITLE
feat(codes): surface error codes

### PR DESCRIPTION
**Pull request by ajpawlicki**
## What
Surfaces error codes if they exist and if `permeatErrorCode` option is set to true

## Why
To surface error codes

## Related PRs
lob/create-boom-error#4

## Next steps
* Cut a new major version for this pkg
* Update to new version in lob/lob-api

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -106,10 +104,8 @@ exports.register = function (server, options, next) {
       }
     };
 
-    if (options.permeateErrorName && err.name) {
-      err.output.payload.error.type = options.decamelizeErrorName ?
-        Decamelize(err.name) : // method_not_allowed
-        err.name; // MethodNotAllowed
+    if (options.permeateErrorCode && err.code) {
```
Comment (ajpawlicki) [inline lib/index.js:107]: I'm adding the `permeateErrorCode` flag so we can release the entire error codes feature by just switching the flag on

File: package.json [hunk 2]
```diff
@@ -25,8 +25,8 @@
     "sinon": "^1.17.3"
   },
   "peerDependencies": {
-    "joi": ">= 10.0.0",
-    "hapi": ">= 16.0.0"
+    "hapi": ">= 16.0.0",
```
Comment (tlhunter) [inline package.json:28]: This will be a SemVer Major change

**Comment by tlhunter** [comment]
Can you also update README.md with the new docs?

**Comment by ajpawlicki** [comment]
@tlhunter readme updated!

**Review by tlhunter — APPROVED** [review]
LGTM

**Pull request merged**